sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/UnwrapCastInBinaryComparison.scala

```diff
@@ -452,9 +452,7 @@ object UnwrapCastInBinaryComparison extends Rule[LogicalPlan] {
       fromExp: Expression,
       toType: DataType,
       literalType: DataType): Boolean = {
-    DataTypeUtils.sameType(toType, literalType) &&
-      !fromExp.foldable &&
-      toType.isInstanceOf[NumericType] &&
+    DataTypeUtils.sameType(toType, literalType) && !fromExp.foldable &&
```

Comment:
Hmm, although currently all callers's `toType` are `NumericType`, it doesn't prevent future caller to pass other types. I guess it is used to prevent that?

Reply:
Ya, I agree with @viirya 's comment. 

We had better keep the original code because the method description explicitly says.
> Note this only handles the case when both sides are of numeric type.